php/libraries/Candidate.class.inc

```diff
@@ -98,7 +98,8 @@ class Candidate
 
         if (!is_array($row) || sizeof($row) == 0) {
             throw new LorisException(
-                "Could not select Candidate data from the database (DCCID: $candID)",
+                'Could not select Candidate data from the database (DCCID: '
+                                                . htmlspecialchars($candID),
```

Comment:
I don't think it makes sense to do `htmlspecialchars` here. The exception message isn't necessarily being printed in an HTML context (for instance, a script might throw an exception.)

The appropriate escaping for the context should be done before being printed, not when the error string is generated.